cmd/utils/flags.go

```diff
@@ -776,16 +776,20 @@ func SetupNetwork(ctx *cli.Context) {
 	params.TargetGasLimit = common.String2Big(ctx.GlobalString(TargetGasLimitFlag.Name))
 }
 
-// MustMakeChainConfig reads the chain configuration from the given database.
+// MustMakeChainConfig reads the chain configuration from the database in ctx.Datadir.
 func MustMakeChainConfig(ctx *cli.Context) *core.ChainConfig {
-	var (
-		db      = MakeChainDatabase(ctx)
-		genesis = core.GetBlock(db, core.GetCanonicalHash(db, 0))
-	)
+	db := MakeChainDatabase(ctx)
 	defer db.Close()
 
+	return MustMakeChainConfigFromDb(ctx, db)
+}
+
+// MustMakeChainConfigFromDb reads the chain configuration from the given database.
+func MustMakeChainConfigFromDb(ctx *cli.Context, db ethdb.Database) *core.ChainConfig {
```

Comment:
The returned `ChainConfig` contains the homestead block number. `ctx` is used to determine if we are on the test or live net which have different homestead block numbers.